pytorch_lightning/trainer/trainer.py

```diff
@@ -903,11 +904,16 @@ def _run(self, model: 'pl.LightningModule') -> Optional[Union[_EVALUATE_OUTPUT,
 
     def _pre_dispatch(self):
         self.accelerator.pre_dispatch(self)
+        self._log_hyperparams()
 
+    def _log_hyperparams(self):
         # log hyper-parameters
         if self.logger is not None:
             # save exp to get started (this is where the first experiment logs are written)
-            self.logger.log_hyperparams(self.lightning_module.hparams_initial)
+            datamodule_hparams = self.datamodule.hparams_initial if self.datamodule is not None else {}
+            hparams_initial = merge_hparams(self.lightning_module.hparams_initial, datamodule_hparams)
```

Comment:
Not sure if having the `merge_hparams` is useful considering how simple it is to merge two dictionaries in python

Reply:
Removed & updated!